scrapy/utils/signal.py

```diff
@@ -1,118 +1,37 @@
-"""Helper functions for working with signals"""
+import warnings
 
-from __future__ import annotations
 
-import logging
-from collections.abc import Sequence
-from typing import Any as TypingAny
-
-from pydispatch.dispatcher import (
-    Anonymous,
-    Any,
-    disconnect,
-    getAllReceivers,
-    liveReceivers,
-)
-from pydispatch.robustapply import robustApply
-from twisted.internet.defer import Deferred, DeferredList
-from twisted.python.failure import Failure
-
-from scrapy.exceptions import StopDownload
-from scrapy.utils.defer import maybeDeferred_coro
-from scrapy.utils.log import failure_to_exc_info
-
-logger = logging.getLogger(__name__)
+def send_catch_log(*args, **kwargs):
+    warnings.warn(
```

Comment:
I think instead of doing this in every function you can do this at the top level like in https://github.com/scrapy/scrapy/blob/master/scrapy/spidermiddlewares/offsite.py and just re=export functions from `signal_ops` instead of calling them.